### PR TITLE
Convert propTypes to Flow Props

### PR DESCRIPTION
**Pull request by AgtLucas**
Associated Issue: #2804 

### Summary of Changes

* Convert `App` and `WelcomeBox` to Flow Props

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] yarn test
- [x] yarn run flow

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -37,12 +37,21 @@ const WelcomeBox = createFactory(_WelcomeBox);
 import _EditorTabs from "./Editor/Tabs";
 const EditorTabs = createFactory(_EditorTabs);
 
+type Props = {
+  selectSource: Function,
+  selectedSource: Object,
```
Comment (jasonLaster) [inline src/components/App.js]: Lets use the `Source` or `SourceRecord` type here

Comment (AgtLucas) [inline src/components/App.js]: @jasonLaster Ok! :)

**Comment by jasonLaster** [comment]
Thanks @AgtLucas!

**Comment by AgtLucas** [comment]
@jasonLaster No problem! :)

**Pull request merged**